client/transaction-pool/graph/src/pool.rs

```diff
@@ -68,6 +68,8 @@ pub trait ChainApi: Send + Sync {
 	type Error: From<error::Error> + error::IntoPoolError;
 	/// Validate transaction future.
 	type ValidationFuture: Future<Output=Result<TransactionValidity, Self::Error>> + Send + Unpin;
+	/// Body future (since block body might be remote)
+	type BodyFuture: Future<Output = Result<Option<Vec<<Self::Block as traits::Block>::Extrinsic>>, Self::Error>> + Unpin + Send + 'static;
```

Comment:
What's the difference between `None` and `Some(Vec::new())` - probably just keep it as `Vec<...>`?

Reply:
Better probably follow what client trait `BlockBody` returns, it is `Result<Option<Vec<Extrinsic>>>`, what do you think?